.github/workflows/test.yml

```
@@ -1,68 +1,114 @@
 on:
-  # Triggers the workflow on push or pull request events but only for the main branch
   push:
     branches:
       - main
   pull_request:
-  # Allows you to run this workflow manually from the Actions tab
   workflow_dispatch:
-
-# A workflow run is made up of one or more jobs that can run sequentially or in parallel
 jobs:
-  test-flow:
+  jest:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        node: [10, 12, 14, 15]
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+      - name: Cache node modules
+        uses: actions/cache@v2
+        env:
+          cache-name: cache-node-modules
+        with:
+          # npm cache files are stored in `~/.npm` on Linux/macOS
+          path: ~/.npm
+          key: ${{ runner.os }}-build-${{ env.cache-name }}-${{ matrix.node }}-${{ hashFiles('**/package-lock.json') }}
+          restore-keys: |
+            ${{ runner.os }}-build-${{ env.cache-name }}-${{ matrix.node }}-
+      - name: Load Node version ${{ matrix.node }}
+        uses: actions/setup-node@v2
+        with:
+          node-version: ${{ matrix.node }}
+      - name: Npm Install
+        run: npm install
+      - name: Run Jest
+        run: npm run test:ci
+      - name: Run Coveralls
+        uses: coverallsapp/github-action@v1.1.2
+        with:
+          github-token: ${{ secrets.GITHUB_TOKEN }}
+          path-to-lcov: ./reports/lcov.info
+  flow:
     runs-on: ubuntu-latest
     steps:
         # Checkss out the repository under $GITHUB_WORKSPACE
       - name: Checkout
         uses: actions/checkout@v2
+      - name: Cache node modules
+        uses: actions/cache@v2
+        env:
+          cache-name: cache-node-modules
+        with:
+          # npm cache files are stored in `~/.npm` on Linux/macOS
+          path: ~/.npm
+          key: ${{ runner.os }}-build-${{ env.cache-name }}-${{ hashFiles('**/package-lock.json') }}
+          restore-keys: |
+            ${{ runner.os }}-build-${{ env.cache-name }}-
+            ${{ runner.os }}-build-
+            ${{ runner.os }}-
       - name: Load Node
-        uses: actions/setup-node@v1
+        uses: actions/setup-node@v2
       - name: Npm Install
         run: npm install
       - name: Flow type check
         run: npm run flow
-  test-eslint:
-    # The type of runner that the job will run on
+  eslint:
     runs-on: ubuntu-latest
     strategy:
       matrix:
-        eslint: [3, 4, 5, 6]
-        node: [6, 7, 8, 9, 10, 12]
-        exclude:
-          - node: 7
-            eslint: 6
-          - node: 7
-            eslint: 5
-          - node: 7
-            eslint: 4
-          - node: 7
-            eslint: 3
-          - node: 6
-            eslint: 6
-          - node: 6
-            eslint: 5
-          - node: 6
-            eslint: 4
-          - node: 6
-            eslint: 3
-
-    # Steps represent a sequence of tasks that will be executed as part of the job
+        node: [10, 12, 14, 15]
```

12 is the current LTS, and you probably don't need to run ESLint a bunch of times
```suggestion
        node: [12]
```